REACT_STYLEGUIDE.MD

```diff
@@ -45,14 +44,31 @@ Bad:
     };
   },
   onClick() {
-    this.setState({ clickCount: this.state.clickCount++ });
+    this.setState(prevState => ({
+      clickCount: prevState.clickCount + 1
+    }));
   },
-  render({ onClick }) {
+  render() {
     return <button className="kuiButton" onClick={this.onClick}/>
   }
 });
 ```
 
+### When a state change only involves the previous state, pass setState a function instead of an object.
```

Comment:
I added this due to the comment from @kjbekkelund so he may be better able to respond, but after reading the linked docs, I believe if you are making a change that doesn't require previousState, then it's simpler to pass in an object.  But I did word that strangely, so I updated it so that if it ever involves the previous state (not "only"), use a function instead of an object.

```
this.setState({ quantity: 2 })
```
vs
```
this.setState(() => ({ quantity: 2 }))
```